Feature/mirror upgrade

lukasgolson opened the pull request:
resolved #71 

**Needs to be tested more; I've tested it locally but I haven't made sure that PR behavior matches pre-PR behavior.**

This brings an increase in maintainability and performance.

Commit:
Updated SyncDictionary usage in SyncedBag.cs

- Removed the extra class that is no longer needed for SyncDictionary's
- Removed OP codes (OP_DIRTY) switch case that no longer exists (superseded by OP_SET)

Signed-off-by: Lukas Olson <lukasolson@greyblockgames.com>

Commit:
Updated method names

Signed-off-by: Lukas Olson <lukasolson@greyblockgames.com>

Commit:
Updated more string sync dictionaries

Ped_Networking.cs
VehicleController.cs
Player.cs
SyncedServerData.cs

Signed-off-by: Lukas Olson <lukasolson@greyblockgames.com>

Commit:
Updated NetworkTime fields in NetStats.cs

Signed-off-by: Lukas Olson <lukasolson@greyblockgames.com>

Commit:
Updated syncData types

Signed-off-by: Lukas Olson <lukasolson@greyblockgames.com>

Commit:
Implemented conditional compilation to replace old isHeadless

Signed-off-by: Lukas Olson <lukasolson@greyblockgames.com>

Commit:
Updated hooks

Commit:
A few more syncDictionary upgrades

Commit:
Moved away from obsolete NetworkIdentity.spawned

Commit:
Updated SetDirtyBit to SetSyncVarDirtyBit in DeadBody.cs

Commit:
Updated ScriptingDefineSymbols for Mirror

Commit:
Updated JoinGameWindow.cs

Commit:
Use latest MirrorLite commit for submodule

in0finite commented on Assets/Scripts/Networking/NetCmdLineHandler.cs:
```diff
@@ -8,25 +8,26 @@ namespace SanAndreasUnity.Net
         IEnumerator Start()
         {
-            if (!Mirror.NetworkManager.isHeadless)
+#if UNITY_SERVER
```
I think this is not the right approach - the application can run in headless mode, even if it's not built with `Server build` flag.

Instead, use `F.IsInHeadlessMode`.

in0finite commented on Assets/Scripts/Networking/Player.cs:
```diff
@@ -113,16 +113,16 @@ void Start()
         }
 
-        public override void OnNetworkDestroy()
+        public override void OnStopClient()
```
will this be executed on server also ?

lukasgolson replied:
It's my understanding that it's never been called on the server. _OnNetworkDestroy_ was renamed to _OnStopClient_ to reduce user confusion. Have we been expecting it to be called on the server?

See: https://mirror-networking.gitbook.io/docs/general/deprecations#networkbehaviour

in0finite replied:
hmm, I remember seeing this log message on server also, but anyway it won't change anything so we are fine.

in0finite commented on Assets/Scripts/Networking/SyncedBag.cs:
```diff
@@ -27,24 +25,22 @@ public ArrayWrapper(T[] a)
             {
-                case StringSyncDictionary.Operation.OP_ADD:
-                case StringSyncDictionary.Operation.OP_SET:
-                case StringSyncDictionary.Operation.OP_DIRTY:
```
is `OP_DIRTY` removed ?

lukasgolson replied:
Yup! https://mirror-networking.gitbook.io/docs/general/deprecations#synclist-operations

in0finite commented on Assets/Scripts/UI/JoinGameWindow.cs:
```diff
@@ -108,132 +106,136 @@ protected override void OnWindowGUI ()
-					buttonAction = async () => await RefreshMasterServersButtonPressed();
+                    //		GUI.enabled = ! m_netDiscoveryHUD.IsRefreshing;
+                    //		buttonText = m_netDiscoveryHUD.IsRefreshing ? ( "Refreshing." + new string('.', (int) ((Time.time * 2) % 3)) ) : "Refresh LAN";
+                    //		buttonAction = () => m_netDiscoveryHUD.Refresh();
```
what happened to this, is LAN functionality still available ?

lukasgolson replied:
In my testing it was, but it looks like I hadn't finished working on this part; I'll clean this up and get back to you!

in0finite commented on Assets/Scripts/UI/JoinGameWindow.cs:
```diff
@@ -4,102 +4,100 @@
 using SanAndreasUnity.Net;
```
please revert the entire file, and only change what is necessary, disable auto-format in Rider like I mentioned in the comment

in0finite commented on ProjectSettings/EditorSettings.asset:
```diff
@@ -3,19 +3,38 @@
 --- !u!159 &1
```
which version of Unity you use ?

Is there anything changed in this file or is it only re-generated by Unity ?

lukasgolson replied:
This change can be discarded, it was included as an oversight. I'll fix this.

in0finite commented:
Please disable auto-formatting in Rider, it's creating a mess in PR, and it becomes more difficult to review it.

in0finite commented on Assets/Scripts/Networking/CustomNetworkManager.cs:
```diff
@@ -59,26 +56,20 @@ void OnLoaderFinished()
             }
         }
-
-        public override void ConfigureServerFrameRate()
```
from Mirror docs: ConfigureServerFrameRate was renamed to ConfigureHeadlessFrameRate.

so we need to override it, or otherwise it will assign frame rate

lukasgolson committed:
Use latest MirrorLite commit for submodule

lukasgolson committed:
Reverted EditorSettings.asset to commit b1c9d38e3ac5113e899ac34ad07bcb8a2890d18b

lukasgolson committed:
Reverted JoinGameWindow.cs to commit b1c9d38e3ac5113e899ac34ad07bcb8a2890d18b

lukasgolson committed:
Changed method for headless mode in NetCmdLineHandler.cs

Changed from compiler defs to SanAndreasUnity helpers for determining headless mode in NetCmdLineHandler.cs

lukasgolson committed:
Re-Added ConfigureHeadlessFrameRate override on CustomNetworkManager.cs

lukasgolson committed:
Started updating JoinGameWindow.cs

- Commented out GUI errors
- Updated type 'DiscoveryInfo' to 'ServerResponse' in method ConnectFromDiscovery() params.
- Updated Connect() 'port' parameter to use type 'int' rather than 'ushort' as per Mirror conventions.
